### PR TITLE
refactor: remove chain id from index for observer set (cherry picked)

### PR DESCRIPTION
**Pull request by lumtis**
# Description

From @kingpinXD 

- Rename ObserverMapper to ObserverSet
- Removes chain id from index for ObserverSet

### CODE REVIEW

File: app/setup_handlers.go [hunk 1]
```diff
@@ -17,7 +18,8 @@ func SetupHandlers(app *App) {
 		for m, mb := range app.mm.Modules {
 			vm[m] = mb.ConsensusVersion()
 		}
-		vm[crosschaintypes.ModuleName] = vm[crosschaintypes.ModuleName] - 1
+		vm[crosschaintypes.ModuleName] = 3
+		vm[observertypes.ModuleName] = 4
```
Comment (kingpinXD) [inline app/setup_handlers.go:22]: c03376a7f521c7860f7d73f173222123021ac0bd

**Review by kingpinXD — APPROVED** [review]

File: x/crosschain/types/expected_keepers.go [hunk 2]
```diff
@@ -44,10 +44,9 @@ type BankKeeper interface {
 	MintCoins(ctx sdk.Context, moduleName string, amt sdk.Coins) error
 }
 
-type ZetaObserverKeeper interface {
-	SetObserverMapper(ctx sdk.Context, om *observertypes.ObserverMapper)
-	GetObserverMapper(ctx sdk.Context, chain *common.Chain) (val observertypes.ObserverMapper, found bool)
-	GetAllObserverMappers(ctx sdk.Context) (mappers []*observertypes.ObserverMapper)
+type ObserverKeeper interface {
+	SetObservers(ctx sdk.Context, om observertypes.ObserverSet)
```
Comment (brewmaster012) [inline x/crosschain/types/expected_keepers.go:48]: minor: if correct, to make it consistent with below, 
`SetObserverSet`?

**Review by brewmaster012 — APPROVED** [review]

File: x/crosschain/types/expected_keepers.go (hunk 2 above)
Comment (kingpinXD) [reply x/crosschain/types/expected_keepers.go:48]: I did not use the name as it sounds a bit weird to  Set-Observer-Set , but happy to make the change if it sounds okay to everyone else.

Comment (lumtis) [reply x/crosschain/types/expected_keepers.go:48]: @kingpinXD I added the change, I will push it to the branch

Comment (lumtis) [reply x/crosschain/types/expected_keepers.go]: [4481ff1](https://github.com/zeta-chain/node/pull/1440/commits/4481ff1bca76ebf5af9859b0dfbe7dcbcc8cdca4)

**Pull request merged**